include/aspect/heating_model/interface.h

```diff
@@ -298,6 +323,30 @@ namespace aspect
     };
 
 
+
+    /**
+     * Go through the list of all heating models that have been selected in
+     * the input file (and are consequently currently active) and see if one
+     * of them has the desired type specified by the template argument. If so,
+     * return a pointer to it. If no heating model is active that matches the
+     * given type, return a NULL pointer.
+     */
+    template <int dim>
+    template <typename HeatingModelType>
+    inline
+    HeatingModelType *
+    Manager<dim>::find_heating_model () const
+    {
+      for (typename std::list<std_cxx11::shared_ptr<Interface<dim> > >::const_iterator
+           p = heating_model_objects.begin();
+           p != heating_model_objects.end(); ++p)
+        if (HeatingModelType *x = dynamic_cast<HeatingModelType *> ( (*p).get()) )
+          return x;
+      return NULL;
+    }
```

Comment:
Nice. If you're enterprising, the changes to this file could also easily go into a separate patch.